aws-codeguruprofiler-profilinggroup/src/test/java/software/amazon/codeguruprofiler/profilinggroup/CreateHandlerTest.java

```diff
@@ -75,24 +82,84 @@
             .actionGroup(AGENT_PERMISSIONS)
             .principals(principals)
             .build();
+    private final AddNotificationChannelsRequest addNotificationChannelsRequest = AddNotificationChannelsRequest.builder()
+            .profilingGroupName(profilingGroupName)
+            .channels(Channel.builder()
+                    .uri("channelUri")
+                    .eventPublishers(ImmutableSet.of(EventPublisher.ANOMALY_DETECTION))
+                    .build())
+            .build();
 
     private final DeleteProfilingGroupRequest deleteProfilingGroupRequest = DeleteProfilingGroupRequest.builder()
             .profilingGroupName(profilingGroupName)
             .build();
 
     @Nested
-    class WhenPermissionsAreNotSet {
-
+    class WhenAnomalyDetectionNotificationConfigurationIsSet {
         @BeforeEach
         public void setup() {
-            request = makeRequest(ResourceModel.builder().profilingGroupName(profilingGroupName).build());
+            request = makeRequest(ResourceModel.builder().profilingGroupName(profilingGroupName)
+                    .anomalyDetectionNotificationConfiguration(AnomalyDetectionNotificationConfiguration.builder()
+                            .channels(Collections.singletonList(software.amazon.codeguruprofiler.profilinggroup.Channel.builder()
+                                    .channelUri("channelUri")
+                                    .build()
+                            ))
+                            .build())
+                    .build());
 
             doReturn(CreateProfilingGroupResponse.builder()
-                         .profilingGroup(ProfilingGroupDescription.builder().name(profilingGroupName).build())
-                         .build())
+                    .profilingGroup(ProfilingGroupDescription.builder().name(profilingGroupName).build())
+                    .build())
                     .when(proxy).injectCredentialsAndInvokeV2(eq(createPgRequest), any());
         }
 
+        @Test
+        public void itShouldDeleteProfilingGroupWhenAddNotificationChannelFails() {
+            when(proxy.injectCredentialsAndInvokeV2(eq(addNotificationChannelsRequest), any())).thenThrow(InternalServerException.class);
+            assertThrows(CfnServiceInternalErrorException.class,
+                                () -> subject.handleRequest(proxy, request, null, logger));
+
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(createPgRequest), any());
+            verify(proxy, times(1)).injectCredentialsAndInvokeV2(eq(addNotificationChannelsRequest), any());
```

Comment:
These checks don't seem to add a lot -- what we're testing here is that when the add fails, we trigger a delete, so this doesn't seem to add a lot.

I suggest creating a separate test to validate that `addNotificationChannels` is correctly called.